core/composition.js

```diff
@@ -0,0 +1,47 @@
+import Embed from '../blots/embed';
+import Emitter from './emitter';
+
+class Composition {
+  constructor(scroll, emitter) {
+    this.scroll = scroll;
+    this.emitter = emitter;
+    this.isComposing = false;
+    scroll.domNode.addEventListener('compositionstart', (event) => {
+      if (!this.isComposing) {
+        this.handleCompositionStart(event);
+      }
+    });
+
+    scroll.domNode.addEventListener('compositionend', (event) => {
+      if (this.isComposing) {
+        this.handleCompositionEnd(event);
+      }
+    });
+  }
+
+  isConposingStagted() {
```

Comment:
something went wrong here :) 
Maybe we can just name internal prop as `_isComposing` and rename this method with `isComposing`?

Reply:
we can't do it without changing of lint rules. At present we have enabled 'no-underscore-dangle'. I'm not sure what we should change this rule...
BTW, method has typo. I will fix it anyway.

Reply:
then i suggest `isCompositionInProgress`

Reply:
renamed